src/components/resource-picker/storage-location.js

```diff
@@ -123,29 +79,38 @@ function StorageLocation({
 
   const getFilesViewerClasses = isDragActive => classNames({
     'StorageLocation-filesViewer': true,
-    'u-can-drop': isDragActive && canAcceptFiles,
-    'u-cannot-drop': isDragActive && !canAcceptFiles
+    'u-can-drop': isDragActive && !isLoading,
+    'u-cannot-drop': isDragActive && !!isLoading
   });
 
+  const FilesViewer = filesViewerDisplay === FILES_VIEWER_DISPLAY.grid
+    ? FilesGridViewer
+    : FilesListViewer;
+
   return (
     <div className="StorageLocation">
       <div className="StorageLocation-buttonsLine">
         <div className="StorageLocation-buttonsLineItem">
-          <DebouncedInput
-            elementType={Search}
-            placeholder={t('common:search')}
-            value={typedInSearchTerm}
-            onSearch={handleSearchClick}
-            onChange={handleSearchTermChange}
+          <Input
+            allowClear
+            value={filterText}
+            placeholder={t('filterPlaceholder')}
+            onChange={handleFilterTextChange}
```

Comment:
One thing I did in my current ticket (which has a real-time table filter behaving in the exact same way) was adding `prefix={<SearchOutlined />}` which looks nice (I know it's technically no search, but people will use it to search for something). Maybe we could align on it (or something similar)?

Reply:
Good idea, added it too 👍 
I tried using the AntD `FilterOutlined` icon but it does not look good enough to make it intuitive:
<img width="409" alt="Screen Shot 2023-01-26 at 16 06 22" src="https://user-images.githubusercontent.com/8189923/214872055-d505347e-79e7-452d-b3ac-4645f6ac22ac.png">

Reply:
That's exactly what I tried too but realized I perceive it as a speaker box that some rotated :stuck_out_tongue_closed_eyes: